qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/AllergiesSteps.java

```diff
@@ -65,4 +81,71 @@ public void removeNoKnownAllergy() {
 	public void systemRemovesNoKnownAllergies() {
 		assertTrue(textExists("Unknown"));
 	}
+	
+	@When("a user clicks Add New Allergy button")
+	public void loadAddAllergyPage() {
+		addOrEditAllergyPage = allergyPage.clickOnAddNewAllergy();
+	}
+	
+	@Then("the system loads add new allergy page")
+	public void systemLoadsAddAllergyPage() {
+		assertTrue(textExists("Add New Allergy"));
+	}
+	
+	@And("a user selects an allergy")
+	public void selectKnownAllergy() {
+		addOrEditAllergyPage.enterDrug(DRUG_NAME);
+		addOrEditAllergyPage.drugId();
+		addOrEditAllergyPage.enterReaction(REACTION);
+		addOrEditAllergyPage.reactionId();
+		addOrEditAllergyPage.addAllergyNote(ALLERGY_NOTE);
+	}
+	
+	@And("a user clicks on the save allergy button")
+	public void saveAllergy() {
+		addOrEditAllergyPage.clickOnSaveAllergy();
+	}
+	
+	@Then("the system adds known allergy into allergies table")
+	public void systemAddsKnownAllergy() {
+		assertTrue(allergyPage.getAllergen().contains(DRUG_NAME));
+	}
+	
+	@When("a user clicks on the edit Allergy icon")
+	public void loadEditAllergyPage() {
+		addOrEditAllergyPage = allergyPage.clickOnEditAllergy();
+	}
+	
+	@Then("the system loads edit allergy page")
+	public void systemLoadsEditAllergyPage() {
+		assertTrue(textExists("Edit Allergy"));
+	}
+	
+	@And("a user edits an allergy")
+	public void editKnownAllergy() {
+		addOrEditAllergyPage.enterReaction(NEW_REACTION);
+		addOrEditAllergyPage.reactionId();
+		addOrEditAllergyPage.addAllergyNote(NEW_ALLERGY_NOTE);
+	}
+	
+	@Then("the system adds edited allergy into the allergies table")
+	public void systemAddsEditedAllergy() {
+		assertTrue(allergyPage.getReaction().contains(NEW_REACTION));
+	}
+		
+	@When("a user clicks on the delete Allergy icon")
+	public void clickOnDeleteAllergyIcon() {
+		allergyPage.clickOnDeleteAllergy();
+	}
+	
+	@Then("a user confirms the delete action")
+	public void clickOnConfirmDeleteAllergy() {
+		allergyPage.clickOnConfirmDeleteAllergy();
+	}
+	
+	@And("the system deletes an allergy from the allergies table")
+	public void systemDeletesAllergy() {
+		assertTrue(allergyPage.getAllergyStatus().contains("Unknown"));
```

Comment:
what is the test asserting here?

Reply:
> what is the test asserting here?

@kdaud  when there's no allergy captured, the allergy table has that word Unknown displayed in the table. So the assertion is to prove that that word exists on the allergy page.

Reply:
why not assert that the allergy table is empty?

Reply:
> why not assert that the allergy table is empty after deleting the allergy?

@kdaud then that means I need to introduce that method in the allergy page because it does not exist. If I am to go through the  addOrEditAllergyPage using this `assertNull(addOrEditAllergyPage.getAllergiesList());` the test hangs until if fails with `Caused by: java.lang.RuntimeException: NettyHttpHandler request execution error` . Also I have tried to assert false to the existence of the allergen after deleting it and still it's failing `assertFalse(allergyPage.getAllergen().contains(DRUG_NAME));`, I would prefer the last approach since if confirms that the deleted allergen is no more existing on the  allergy page. Any thoughts?

Reply:
asserting that the allergy table is empty after deleting the allergy would be the better ideal.

Reply:
> asserting that the allergy table is empty after deleting the allergy would be the better ideal.

@kdaud did you take a look at the error logs I share in the[ above comment ](https://github.com/openmrs/openmrs-contrib-qaframework/pull/298#discussion_r782834515)? Like i said if I implement assertNull(addOrEditAllergyPage.getAllergiesList());  the test keeps hanging and finally breaks. Do you have an idea? Could you try running it on your side and help me find out the result.

Reply:
@Irenyak1 the ideal is to have a better implementation of the test. I  would think when the allergy is deleted from the allergen table, then asserting that the table is null is a better approach for the test.

Reply:
> @Irenyak1 the ideal is to have a better implementation of the test. I would think when the allergy is deleted from the allergen table, then asserting that the table is null is a better approach for the test.

@kdaud I am not object the better implementation at all. My request to you was to help me run the test on your side and give me the feedback. Is what you are talking about different from this `assertNull(addOrEditAllergyPage.getAllergiesList());`?

Reply:
> My request to you was to help me run the test on your side and give me the feedback

@Irenyak1 the changes in this commit are not yet in master so I probably do not have them locally. However, you need to figure out why the test is failing so that we have a better implementation of the workflow.

Reply:
@Irenyak1  i can see the test passing. is there anything different for @Irenyak1  to address ?

Reply:
@sherrif10 you may need to follow up the conversation from its [root](https://github.com/openmrs/openmrs-contrib-qaframework/pull/298#discussion_r782719576).

Reply:
Reviewed all the comments. @Irenyak1  are you able to respond to the above reviews.

Reply:
> why not assert that the allergy table is empty after deleting the allergy?

@kdaud  To make things simple and precise. 

> assertTrue(allergyPage.getAllergyStatus().contains("Unknown"));  fits the solution. What you are trying to mean is the same as indicated here using the contains("Unknown) texts unless you recommend her the suggestion such that he can apply. it seems this is delaying her to progress with other tasks.